src/thor/route_action.cc

```diff
@@ -540,7 +561,8 @@ void thor_worker_t::path_depart_at(Api& api, const std::string& costing) {
       if (!origin->date_time().empty() && options.date_time_type() != valhalla::Options::invariant) {
         auto destination_dt =
             offset_date(*reader, origin->date_time(), temp_path.front().edgeid,
-                        temp_path.back().elapsed_cost.secs, temp_path.back().edgeid);
+                        temp_path.back().elapsed_cost.secs + destination->waiting_secs(),
+                        temp_path.back().edgeid);
```

Comment:
Here we're in the `depart_at`'s `route_two_location` lamdba and things are much easier since we're going forward, not reverse. We just have to add `waiting_secs` to the elapsed time. No other special treatment necessary.